### ISSUE TITLE
Hunt-Framework broken with Poodinis 8.1.0-beta.3

### ISSUE THREAD

**Issue by Heromyth** [mbierlee/poodinis#32]
It's so sad that the Hunt-Framework has broken with Poodinis 8.1.0-beta.3. 
When you see this message, we have forced it to use Poodinis 8.1.0-beta.2.

Here are the errors:
```sh
hunt-framework 3.4.0-beta.3+commit.21.g7877f83: building configuration "library"...
../../poodinis/source/poodinis/container.d(414,28): Error: template instance `std.traits.isFunction!(BaseObject, Object)` does not match template declaration `isFunction(X...)`
  with `X = (hunt.pool.BaseObject.BaseObject, object.Object)`
  must satisfy the following constraint:
`       X.length == 1`
../../poodinis/source/poodinis/container.d(331,33): Error: template instance `poodinis.container.DependencyContainer.callPostConstructors!(RedisPoolConfig)` error instantiating
../../poodinis/source/poodinis/container.d(275,60):        instantiated from here: `resolve!(RedisPoolConfig, RedisPoolConfig)`
../../poodinis/source/poodinis/factory.d-mixin-162(163,80):        instantiated from here: `resolve!(RedisPoolConfig)`
../../poodinis/source/poodinis/container.d(183,36):        instantiated from here: `ConstructorInjectingInstanceFactory!(RedisPool)`
../../poodinis/source/poodinis/container.d(314,59):        ... (1 instantiations, -v to show) ...
../../poodinis/source/poodinis/container.d(275,60):        instantiated from here: `resolve!(RedisPool, RedisPool)`
../../hunt-framework/source/hunt/framework/application/Application.d(509,60):        instantiated from here: `resolve!(RedisPool)`
../../poodinis/source/poodinis/autowire.d(273,28): Error: template instance `std.traits.isFunction!(BaseObject, Object)` does not match template declaration `isFunction(X...)`
  with `X = (hunt.pool.BaseObject.BaseObject, object.Object)`
  must satisfy the following constraint:
`       X.length == 1`
../../poodinis/source/poodinis/container.d(184,36): Error: template instance `poodinis.autowire.AutowiredRegistration!(RedisPoolConfig)` error instantiating
```

**The `container.register` is the problem.** 

After using `initializedBy`, most of registrations passed except for the `hunt.framework.view.View`.

```
../../poodinis/source/poodinis/autowire.d-mixin-273(273,1): Error: no property `TemplateLexer` for type `hunt.framework.view.Environment.Environment`
../../poodinis/source/poodinis/container.d(184,36): Error: template instance `poodinis.autowire.AutowiredRegistration!(Environment)` error instantiating
../../poodinis/source/poodinis/container.d(148,53):        instantiated from here: `register!(Environment, Environment)`
../../hunt-framework/source/hunt/framework/provider/ViewServiceProvider.d(23,18):        instantiated from here: `register!(Environment)`
../../poodinis/source/poodinis/container.d-mixin-414(414,1): Error: no property `TemplateLexer` for type `hunt.framework.view.Environment.Environment`
../../poodinis/source/poodinis/container.d(331,33): Error: template instance `poodinis.container.DependencyContainer.callPostConstructors!(Environment)` error instantiating
../../poodinis/source/poodinis/container.d(275,60):        instantiated from here: `resolve!(Environment, Environment)`
../../poodinis/source/poodinis/factory.d-mixin-162(163,79):        instantiated from here: `resolve!(Environment)`
../../poodinis/source/poodinis/container.d(183,36):        instantiated from here: `ConstructorInjectingInstanceFactory!(View)`
../../poodinis/source/poodinis/container.d(148,53):        instantiated from here: `register!(View, View)`
../../hunt-framework/source/hunt/framework/provider/ViewServiceProvider.d(25,18):        instantiated from here: `register!(View)`
```

To check this, run them please:
```sh
git clone https://github.com/huntlabs/hunt-skeleton.git myproject
cd myproject
dub run
```

**Comment by mbierlee** [comment]
Thanks for bringing this up. This is why I released another Beta first. I will take a look.

**Comment by mbierlee** [comment]
The root-cause compilation error has been fixed in 8.1.0-beta.4
Some remaining usages of the old register() signature need to be adjusted in Hunt.

**Comment by Heromyth** [comment]
Thanks. Only one problem left, see here https://github.com/mbierlee/poodinis/issues/35.
